trieste/models/keras/sampler.py

```diff
@@ -188,15 +191,25 @@ def resample(self) -> None:
         Efficiently resample network indices in-place, without retracing.
         """
         if self._diversify:
-            indices = tf.random.shuffle(tf.range(self._ensemble_size * self._sample_size))
-            self._indices.assign(indices[: self._batch_size])  # [B]
-            self._seeds.assign(
+            self._indices.assign(
                 tf.random.uniform(
-                    shape=(self._ensemble_size, self._sample_size),
+                    shape=(self._batch_size,),
                     minval=0.000001,
                     maxval=0.999999,
                     dtype=tf.float32,
                 )
-            )  # [E, B/E]
+            )  # [B]
         else:
             self._indices.assign(sample_model_index(self._ensemble_size, self._batch_size))  # [B]
+
+    def get_state(self) -> Dict[str, TensorType]:
```

Comment:
Any reason to having this, rather than say three properties `self.initialized`, `self.batch_size`, `self.indices`?

Reply:
just thinking ahead, so you can get state info from multiple types of samplers with the same method